MeshLib/ElementCoordinatesMappingLocal.cpp

```diff
@@ -0,0 +1,197 @@
+: _pt_translate(std::array<double,3>({{0,0,0}}))
+{
+    //assert (e->getDimension() <= coordinate_system.getDimension());
+
+    // set initial coordinates
+    for(size_t i = 0; i < e->getNNodes(); i++)
+        _point_vec.push_back(MathLib::Point3d(e->getNode(i)->getCoords()));
+
+    //flip(coordinate_system, _point_vec);
+    if (e->getDimension() < coordinate_system.getDimension()) {
+        translate(_point_vec, _point_vec[0]);
+        rotate(*e, coordinate_system, _point_vec);
+    }
+}
+
+void ElementCoordinatesMappingLocal::flip(const CoordinateSystem &coordinate_system,
+                                          std::vector<MathLib::Point3d> &vec_pt)
+{
+    switch(coordinate_system.getType())
+    {
+    case CoordinateSystemType::Y:
+        {
+            for(size_t i = 0; i < vec_pt.size(); i++)
+                std::swap(vec_pt[i][0], vec_pt[i][1]);
+        }
+        break;
+    case CoordinateSystemType::Z:
+        {
+            for(size_t i = 0; i < vec_pt.size(); i++)
+                std::swap(vec_pt[i][0], vec_pt[i][2]);
+        }
+        break;
+    case CoordinateSystemType::XZ:
+        {
+            for(size_t i = 0; i < vec_pt.size(); i++)
+                std::swap(vec_pt[i][1], vec_pt[i][2]);
+        }
+        break;
+    case CoordinateSystemType::YZ:
+        {
+            for(size_t i = 0; i < vec_pt.size(); i++)
+            {
+                double tmp_x = vec_pt[i][0];
+                vec_pt[i][0] = vec_pt[i][1];
+                vec_pt[i][1] = vec_pt[i][2];
+                vec_pt[i][2] = tmp_x;
+            }
+        }
+        break;
+    default:
+        break;
+    }
+}
+
+void ElementCoordinatesMappingLocal::translate(std::vector<MathLib::Point3d> &vec_pt, const MathLib::Point3d origin)
+{
+    for (std::size_t i=0; i<vec_pt.size(); ++i) {
+        vec_pt[i] -= origin;
+    }
+    _pt_translate = origin;
+}
+
+void ElementCoordinatesMappingLocal::rotate(const Element &ele, const CoordinateSystem &coordinate_system, std::vector<MathLib::Point3d> &vec_pt)
+{
+    // compute a rotation matrix
+    getRotationMatrixToOriginal(ele, coordinate_system, vec_pt, _matR2original);
+
+    // rotate the point coordinates
+    const std::size_t global_dim = coordinate_system.getDimension();
+    double const* const coords_node_0 (vec_pt[0].getCoords());
+    Eigen::VectorXd dx = Eigen::VectorXd::Zero(global_dim);
+    Eigen::VectorXd x_new = Eigen::VectorXd::Zero(3);
+    for(std::size_t i = 0; i < ele.getNNodes(); i++)
+    {
+        double const* const coords_node_i (vec_pt[i].getCoords());
+        for (std::size_t j=0; j<global_dim; j++)
+            dx[j] = (coords_node_i[j] - coords_node_0[j]);
+
+        x_new.head(global_dim) = _matR2original.transpose() * dx;
+        _point_vec[i] = MathLib::Point3d(x_new.data());
+    }
+};
+
+// x=Rx' where x is original coordinates and x' is local coordinates
+void ElementCoordinatesMappingLocal::getRotationMatrixToOriginal(
```

Comment:
- What does `ToOriginal`mean? Did you mean origin?
- Is it possible to reuse some of the functions `computeRotationMatrixTo...` and `rotatePoints()` from GeoLib/AnalyticalGeometry.{h,cpp}?